lib/bolt/util.rb

```diff
@@ -118,6 +118,21 @@ def map_vals(hash)
         end
       end
 
+      # Accepts a Data object and returns a copy with all hash keys and values
+      # modified by block.
+      def walk_pairs(data, &block)
```

Comment:
I can't get an intuitive sense of what this function is doing. I think the problem I have with it is that it requires that a particular key will always have the same meaning no matter where it's found in the nesting.